packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-            &hash_util,
-        ])?;
-        let ctx_import = ctx_import.as_array().ok_or_else(|| {
+impl<'s> Embedded<'s> {
+    fn import(
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+    ) -> Result<Self> {
+        fn get_function_from_aray<'s>(
+            scope: &mut v8::HandleScope<'s>,
+            fns: v8::Local<'s, v8::Array>,
+            index: u32,
+        ) -> Result<v8::Local<'s, v8::Function>> {
+            fns.get_index(scope, index)
+                .ok_or_else(|| {
+                    Error::V8(format!("Could not get index {index} from runner.js array"))
+                })?
+                .try_into()
+                .map_err(|err| {
+                    Error::V8(format!(
+                        "Could not convert value at index {index} in runner.js as a function: \
+                         {err}"
+                    ))
+                })
+        }
+
+        let arrow = eval_file(
+            scope,
+            "./src/worker/runner/javascript/apache-arrow-bundle.js",
+        )?;
+        let hash_stdlib = eval_file(scope, "./src/worker/runner/javascript/hash_stdlib.js")?;
+        let hash_util = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/hash_util.js",
+            &[arrow],
+        )?;
+        let batches_prototype = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/batch.js",
+            &[arrow, hash_util],
+        )?;
+
+        let ctx_import = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/context.js",
+            &[hash_util],
+        )?;
+        let ctx_import: v8::Local<'_, v8::Array> = ctx_import.try_into().map_err(|err| {
             Error::FileImport(
-                "./src/worker/runner/javascript/context.js".into(),
-                "Couldn't get array (of functions) from 'context.js'".into(),
+                "./src/worker/runner/javascript/context.js".to_string(),
+                format!("Couldn't get array (of functions) from 'context.js': {err}"),
             )
         })?;
-        let experiment_ctx_prototype = ctx_import.get(0)?;
-        let sim_init_ctx_prototype = ctx_import.get(1)?;
-        let gen_ctx = ctx_import.get(2)?;
-
-        let gen_state = import_file(mv8, "./src/worker/runner/javascript/state.js", vec![
-            &hash_util,
-        ])?;
-        let fns = import_file(mv8, "./src/worker/runner/javascript/runner.js", vec![
-            &arrow,
-            &batches_prototype,
-            &experiment_ctx_prototype,
-            &sim_init_ctx_prototype,
-            &gen_ctx,
-            &gen_state,
-        ])?;
-        let fns = fns.as_array().ok_or_else(|| {
+        let experiment_ctx_prototype = ctx_import
+            .get_index(scope, 0)
+            .ok_or_else(|| Error::V8("Could not get index 0 from context.js array".to_string()))?;
```

Comment:
Can we change these to be clearer, like "Couldn't get experiment_ctx_function from index 0 of the context.js return values"

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2